### ISSUE TITLE
[no-unused-refs] Add ignoring for specific ref names/components

### ISSUE THREAD

**Issue by Ky6uk** [vuejs/eslint-plugin-vue#1476]
**What rule do you want to change?**

`no-unused-refs`

**How will the change be implemented? (New option, new default behavior, etc.)?**

I guess there is should be an option to ignore some component/ref names.

```json
{ "vue/no-unused-refs": ["error", {
  "ignoreComponents": ["ValidationObserver"],
  "ignoreRefNames": ["observer"]
}] }
```

**Please provide some example code that this change will affect:**

In our code we have some components with refs that should be accessed outside of the component itself (from mixins, other components, etc.) Ex: observer component from `vee-validate` plugin.

```vue
<template>
  <ValidationObserver ref="observer" />
</template>
```

```vue
<script lang="ts">
export class ErrorMixin extends Vue {
  $refs!: {
    observer: InstanceType<typeof ValidationObserver>;
  };

  handleError({ message, fields } : ApiErrorModel) : void {
    if (this.$refs.observer !== undefined && fields instanceof Object) {
      this.$refs.observer.setErrors(fields);
    }
  }
}
</script>
```
